### PR TITLE
Refactor HttpHeaders

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- Add more unit tests
- Refactor HttpHeaders
- Minor improvements

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Source/AwsCommonRuntimeKit/http/HttpMessage.swift [hunk 1]
```diff
@@ -29,54 +28,53 @@ public class HttpMessage {
...
+    func getHeader(atIndex index: Int) -> HttpHeader {
         var header = aws_http_header()
-        if aws_http_message_get_header(self.rawValue, &header, index) == AWS_OP_SUCCESS {
-            return HttpHeader(rawValue: header)
+        guard aws_http_message_get_header(self.rawValue, &header, index) == AWS_OP_SUCCESS else {
+            fatalError("Index out of range")
         }
-        return nil
+        return HttpHeader(rawValue: header)
     }
```
Comment (waahm7) [inline Source/AwsCommonRuntimeKit/http/HttpMessage.swift:68]: Crash if index is invalid. This is same as Swift's API https://developer.apple.com/documentation/swift/array/remove(at:)-1p2pj

File: Package.swift [hunk 2]
```diff
@@ -6,7 +6,7 @@ let excludesFromAll = ["tests", "cmake", "CONTRIBUTING.md",
                         "CMakeLists.txt", "README.md"]
 var packageTargets: [Target] = []
 
-var package = Package(name: "AwsCrt",
+var package = Package(name: "aws-crt-swift",
```
Comment (waahm7) [inline Package.swift:9]: Change from #93

File: Source/AwsCommonRuntimeKit/http/HttpRequest.swift [hunk 3]
```diff
@@ -35,21 +35,17 @@ public class HttpRequest: HttpMessage {
     /// Creates an http request which can be passed to a connection.
     /// - Parameters:
     ///   - method: Http method to use. Must be a valid http method and not empty.
-    ///   - path: Path of Http Request. Should not be empty.
+    ///   - path: Path of Http Request. Must not be empty.
```
Comment (graebm) [inline Source/AwsCommonRuntimeKit/http/HttpRequest.swift:38]:
```suggestion
    ///   - path: Path and query string for Http Request. Must not be empty.
```

File: Source/Elasticurl/Elasticurl.swift [hunk 4]
```diff
@@ -272,30 +267,22 @@ struct Elasticurl {
...
-            if headers.add(name: "Host", value: host),
-               headers.add(name: "User-Agent", value: "Elasticurl"),
-               headers.add(name: "Accept", value: "*/*"),
-               headers.add(name: "Swift", value: "Version 5.4") {
-                for header in context.headers {
-                    _ = headers.add(name: header.key, value: header.value)
-                }
-            }
+            var headers = [HttpHeader]()
+            headers.append(HttpHeader(name: "Host", value: host))
```
Comment (graebm) [inline Source/Elasticurl/Elasticurl.swift:271]: trivial: seems like an alternate constructor that accepted `name` and `value` as unnamed args would be nice

**Review by graebm — APPROVED** [review]
trivial fix & ship

File: Source/Elasticurl/Elasticurl.swift (hunk 4 above)
Comment (waahm7) [reply Source/Elasticurl/Elasticurl.swift:271]: Thanks, I have added a addHeader function that takes name & value and removed compression.

**Pull request merged**